src/cuda/image_function_cuda.cu

```diff
@@ -1170,4 +1181,40 @@ namespace Image_Function_Cuda
         launchKernel2D( thresholdCuda, width, height,
                         inY, rowSizeIn, outY, rowSizeOut, width, height, minThreshold, maxThreshold );
     }
+
+    std::vector < uint32_t > ProjectionProfile( const Image & image, bool horizontal )
+    {
+        return Image_Function_Helper::ProjectionProfile( ProjectionProfile, image, horizontal );
+    }
+
+    void ProjectionProfile( const Image & image, bool horizontal, std::vector < uint32_t > & projection )
+    {
+        ProjectionProfile( image, 0, 0, image.width(), image.height(), horizontal, projection );
+    }
+
+    std::vector < uint32_t > ProjectionProfile( const Image & image, uint32_t x, uint32_t y, uint32_t width, uint32_t height, bool horizontal )
+    {
+        return Image_Function_Helper::ProjectionProfile( ProjectionProfile, image, x, y, width, height, horizontal );
+    }
+
+    void Image_Function_Cuda::ProjectionProfile( const Image & image, uint32_t x, uint32_t y, uint32_t width, uint32_t height, bool horizontal,
+                                                 std::vector<uint32_t> & projection )
+    {
+        Image_Function::ParameterValidation( image, x, y, width, height );
+
+        const uint8_t colorCount = image.colorCount();
+
+        projection.resize( horizontal ? width * colorCount : height );
+        std::fill( projection.begin(), projection.end(), 0u );
+
+        const uint32_t rowSize = image.rowSize();
+        width *= colorCount;
+
+        const uint8_t * imageX = image.data() + y * rowSize + x * colorCount;
+
+        multiCuda::Array< uint32_t > projectionCuda( projection );
+
+        launchKernel2D( normalizeCuda, width, height,
+            imageX, rowSize, horizontal, width, height, projectionCuda.data());
+    }
```

Comment:
We should copy `projectionCuda` into `projection` array upon return from this function.